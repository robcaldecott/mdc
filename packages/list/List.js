import React from "react";
import classnames from "classnames";

export const List = ({
  element = "ul",
  nonInteractive,
  dense,
  twoLine,
  className,
  children,
  ...other
}) =>
  React.createElement(
    element,
    {
      className: classnames(
        "mdc-list",
        {
          "mdc-list--non-interactive": nonInteractive,
          "mdc-list--dense": dense,
          "mdc-list--two-line": twoLine
        },
        className
      ),
      ...other
    },
    children
  );
