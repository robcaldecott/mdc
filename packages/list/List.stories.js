import React, { Fragment } from "react";
import {
  List,
  ListItem,
  ListItemText,
  ListItemSecondaryText,
  ListItemGraphic,
  ListItemMeta,
  ListDivider,
  ListGroup,
  ListGroupSubheader,
  ListGroupDivider,
  NavList,
  NavListItem,
  NavListDivider
} from ".";
import { storiesOf } from "@storybook/react";
import { withKnobs, boolean } from "@storybook/addon-knobs";

storiesOf("list", module)
  .addDecorator(withKnobs)
  .add("props", () => {
    const twoLine = boolean("twoLine", false);
    const graphics = boolean("graphics", false);
    const meta = boolean("meta", false);
    const dividers = boolean("dividers", false);
    const padded = boolean("padded dividers", false);
    const inset = boolean("inset dividers", false);
    const selected = boolean("selected items", false);
    const activated = boolean("activate first item", false);
    return (
      <List
        nonInteractive={boolean("nonInteractive", false)}
        dense={boolean("dense", false)}
        twoLine={twoLine}
      >
        {[...Array(5).keys()].map(index => (
          <Fragment>
            <ListItem selected={selected} activated={activated && index === 0}>
              {graphics && (
                <ListItemGraphic>
                  <i className="material-icons">star</i>
                </ListItemGraphic>
              )}
              {!twoLine && `Item ${index + 1}`}
              {twoLine && (
                <ListItemText>
                  Item {index + 1}
                  <ListItemSecondaryText>
                    Secondary item {index + 1}
                  </ListItemSecondaryText>
                </ListItemText>
              )}
              {meta && (
                <ListItemMeta>
                  <i className="material-icons">delete</i>
                </ListItemMeta>
              )}
            </ListItem>
            {dividers && <ListDivider padded={padded} inset={inset} />}
          </Fragment>
        ))}
      </List>
    );
  })
  .add("groups", () => (
    <ListGroup>
      <ListGroupSubheader>List 1</ListGroupSubheader>
      <List>
        <ListItem>Item 1</ListItem>
        <ListItem>Item 2</ListItem>
        <ListItem>Item 3</ListItem>
      </List>
      <ListGroupDivider />
      <ListGroupSubheader>List 2</ListGroupSubheader>
      <List>
        <ListItem>Item 1</ListItem>
        <ListItem>Item 2</ListItem>
        <ListItem>Item 3</ListItem>
      </List>
    </ListGroup>
  ))
  .add("nav list", () => (
    <NavList onClick={e => e.preventDefault()}>
      <NavListItem href="#item1" selected>
        <ListItemGraphic>
          <i className="material-icons">inbox</i>
        </ListItemGraphic>Item 1
      </NavListItem>
      <NavListItem href="#item2">
        <ListItemGraphic>
          <i className="material-icons">star</i>
        </ListItemGraphic>Item 2
      </NavListItem>
      <NavListDivider />
      <NavListItem href="#item3">
        <ListItemGraphic>
          <i className="material-icons">delete</i>
        </ListItemGraphic>Item 3
      </NavListItem>
    </NavList>
  ));
