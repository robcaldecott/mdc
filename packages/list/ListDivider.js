import React from "react";
import { string, bool } from "prop-types";
import classnames from "classnames";

const ListDivider = ({
  element,
  padded,
  inset,
  className,
  children,
  ...other
}) =>
  React.createElement(element, {
    className: classnames(
      "mdc-list-divider",
      {
        "mdc-list-divider--padded": padded,
        "mdc-list-divider--inset": inset
      },
      className
    ),
    role: "separator",
    ...other
  });

ListDivider.propTypes = {
  element: string,
  padded: bool,
  inset: bool
};

ListDivider.defaultProps = {
  element: "li"
};

export default ListDivider;
