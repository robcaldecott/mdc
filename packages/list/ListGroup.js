import React from "react";
import classnames from "classnames";

const ListGroup = ({ className, ...other }) => (
  <div className={classnames("mdc-list-group", className)} {...other} />
);

export default ListGroup;
