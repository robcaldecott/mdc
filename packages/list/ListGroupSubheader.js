import React from "react";
import classnames from "classnames";

const ListGroupSubheader = ({ className, children, ...other }) => (
  <h3 className={classnames("mdc-list-group__subheader", className)} {...other}>
    {children}
  </h3>
);

export default ListGroupSubheader;
