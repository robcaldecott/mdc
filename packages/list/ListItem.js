import React from "react";
import { string, bool } from "prop-types";
import classnames from "classnames";

export const ListItem = ({
  element,
  selected,
  activated,
  className,
  children,
  ...other
}) =>
  React.createElement(
    element,
    {
      className: classnames(
        "mdc-list-item",
        {
          "mdc-list-item--selected": selected,
          "mdc-list-item--activated": activated
        },
        className
      ),
      ...other
    },
    children
  );

ListItem.propTypes = {
  element: string,
  selected: bool,
  activated: bool
};

ListItem.defaultProps = {
  element: "li"
};
