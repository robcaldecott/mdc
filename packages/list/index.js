export { default as List } from "./List";
export { default as ListItem } from "./ListItem";
export { default as ListItemText } from "./ListItemText";
export { default as ListItemSecondaryText } from "./ListItemSecondaryText";
export { default as ListItemGraphic } from "./ListItemGraphic";
export { default as ListItemMeta } from "./ListItemMeta";
export { default as ListDivider } from "./ListDivider";
export { default as ListGroup } from "./ListGroup";
export { default as ListGroupSubheader } from "./ListGroupSubheader";
export { default as ListGroupDivider } from "./ListGroupDivider";
export { default as NavList } from "./NavList";
export { default as NavListItem } from "./NavListItem";
export { default as NavListDivider } from "./ListGroupDivider";
