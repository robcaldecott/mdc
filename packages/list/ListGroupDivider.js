import React from "react";
import ListDivider from "./ListDivider";

const ListGroupDivider = props => <ListDivider element="hr" {...props} />;

export default ListGroupDivider;
