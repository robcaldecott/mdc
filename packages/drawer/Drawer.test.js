import React from "react";
import {
  PermanentDrawer,
  DrawerToolbarSpacer,
  DrawerContent,
  PersistentDrawer,
  DrawerHeader,
  DrawerHeaderContent,
  TemporaryDrawer
} from ".";
import { NavList, NavListItem } from "@mdc/list";
import renderer from "react-test-renderer";

it("renders a permanent drawer", () => {
  expect(
    renderer.create(
      <PermanentDrawer>
        <DrawerToolbarSpacer />
        <DrawerContent>
          <NavList>
            <NavListItem>Item 1</NavListItem>
            <NavListItem>Item 2</NavListItem>
          </NavList>
        </DrawerContent>
      </PermanentDrawer>
    )
  ).toMatchSnapshot();
});

it("renders a persistent drawer", () => {
  expect(
    renderer.create(
      <PersistentDrawer open>
        <DrawerHeader>
          <DrawerHeaderContent>Header</DrawerHeaderContent>
        </DrawerHeader>
        <DrawerContent>
          <NavList>
            <NavListItem>Item 1</NavListItem>
            <NavListItem>Item 2</NavListItem>
          </NavList>
        </DrawerContent>
      </PersistentDrawer>
    )
  ).toMatchSnapshot();
});

it("renders a temporary drawer", () => {
  expect(
    renderer.create(
      <TemporaryDrawer open>
        <DrawerHeader>
          <DrawerHeaderContent>Header</DrawerHeaderContent>
        </DrawerHeader>
        <DrawerContent>
          <NavList>
            <NavListItem>Item 1</NavListItem>
            <NavListItem>Item 2</NavListItem>
          </NavList>
        </DrawerContent>
      </TemporaryDrawer>
    )
  ).toMatchSnapshot();
});
