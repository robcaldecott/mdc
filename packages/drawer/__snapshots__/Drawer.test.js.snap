// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a permanent drawer 1`] = `
<nav
  className="mdc-drawer mdc-drawer--permanent mdc-typography"
>
  <div
    className="mdc-drawer__toolbar-spacer"
  />
  <div
    className="mdc-drawer__content"
  >
    <nav
      className="mdc-list"
    >
      <a
        className="mdc-list-item"
      >
        Item 1
      </a>
      <a
        className="mdc-list-item"
      >
        Item 2
      </a>
    </nav>
  </div>
</nav>
`;

exports[`renders a persistent drawer 1`] = `
<aside
  className="mdc-drawer mdc-drawer--persistent mdc-typography mdc-drawer--open mdc-drawer--animating"
  onTransitionEnd={[Function]}
>
  <nav
    className="mdc-drawer__drawer"
  >
    <header
      className="mdc-drawer__header undefined"
    >
      <div
        className="mdc-drawer__header-content undefined"
      >
        Header
      </div>
    </header>
    <div
      className="mdc-drawer__content"
    >
      <nav
        className="mdc-list"
      >
        <a
          className="mdc-list-item"
        >
          Item 1
        </a>
        <a
          className="mdc-list-item"
        >
          Item 2
        </a>
      </nav>
    </div>
  </nav>
</aside>
`;

exports[`renders a temporary drawer 1`] = `
<aside
  className="mdc-drawer
        mdc-drawer--temporary
        mdc-typography
        undefined
        mdc-drawer--open
        mdc-drawer--animating
        "
  onClick={[Function]}
  onTransitionEnd={[Function]}
>
  <nav
    className="mdc-drawer__drawer"
  >
    <header
      className="mdc-drawer__header undefined"
    >
      <div
        className="mdc-drawer__header-content undefined"
      >
        Header
      </div>
    </header>
    <div
      className="mdc-drawer__content"
    >
      <nav
        className="mdc-list"
      >
        <a
          className="mdc-list-item"
        >
          Item 1
        </a>
        <a
          className="mdc-list-item"
        >
          Item 2
        </a>
      </nav>
    </div>
  </nav>
</aside>
`;
